service/matching/matchingEngine.go

```diff
@@ -548,7 +548,7 @@ func (e *matchingEngineImpl) RespondQueryTaskCompleted(
 func (e *matchingEngineImpl) deliverQueryResult(taskID string, queryResult *queryResult) error {
 	queryResultCh, ok := e.lockableQueryTaskMap.get(taskID)
 	if !ok {
-		return serviceerror.NewInternal("query task not found, or already expired")
+		return serviceerror.NewNotFound("query task not found, or already expired")
```

Comment:
in this case, the query result is not delivered, so the client will timeout on the workflow query?

Reply:
this error is returned by matching to SDK, when SDK trying to respond back, so SDK should stop if query is gone, instead of retry and fail again